File: 백준/Silver/10773. 제로/제로.cc
#include <iostream>
#include <stack>
using namespace std;

int main()
{
	stack<int> stk;
	int k, n;
	int sum = 0;
	cin >> k; //정수의 개수
	for (int i = 0; i < k; i++) {
		cin >> n;
		stk.push(n);
		if (n == 0) {
			stk.pop();
			stk.pop();
		}
	}
	
	while (!stk.empty()) {
		sum += stk.top();
		stk.pop();
	}
	
	cout << sum << endl;
	return 0;
}

File: 백준/Silver/10773. 제로/제로.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Stack;

public class Main {
	
	public static void main(String[] args) throws IOException {		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		Stack<Integer> stack = new Stack<Integer>();
		
		int n = Integer.parseInt(br.readLine());
		
		for (int i = 0; i < n; i++) {
			int k = Integer.parseInt(br.readLine());
			
			if (k == 0) {
				stack.pop();
			} else {
				stack.push(k);
			}
		}
		
		int sum = 0;
		while (!stack.empty()) {
			sum += stack.peek();
			stack.pop();
		}
		System.out.println(sum);
		br.close();
	}
}